conda_smithy/lint_recipe.py

```diff
@@ -1017,15 +1029,393 @@ def sort_osx(versions):
+                parsed_licenses.append(l.license_symbol.key)
+                parsed_exceptions.append(l.exception_symbol.key)
+            else:
+                parsed_licenses.append(l.key)
+    except license_expression.ExpressionError:
+        parsed_licenses = [license]
+
+    licenseref_regex = re.compile(r"^LicenseRef[a-zA-Z0-9\-.]*$")
+    filtered_licenses = []
+    for license in parsed_licenses:
+        if not licenseref_regex.match(license):
+            filtered_licenses.append(license)
+
+    with open(
+        os.path.join(os.path.dirname(__file__), "licenses.txt"), "r"
+    ) as f:
+        expected_licenses = f.readlines()
+        expected_licenses = set([l.strip() for l in expected_licenses])
+    with open(
+        os.path.join(os.path.dirname(__file__), "license_exceptions.txt"), "r"
+    ) as f:
+        expected_exceptions = f.readlines()
+        expected_exceptions = set([l.strip() for l in expected_exceptions])
+    if set(filtered_licenses) - expected_licenses:
+        hints.append(
+            "License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.\n\n"
+            "Documentation on acceptable licenses can be found "
+            "[here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions )."
+        )
+    if set(parsed_exceptions) - expected_exceptions:
+        hints.append(
+            "License exception is not an SPDX exception.\n\n"
+            "Documentation on acceptable licenses can be found "
+            "[here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions )."
+        )
+
+    lints = [lint for lint in lints if lint]
+    hints = [hint for hint in hints if hint]
+
+    return lints, hints
+
+
+def run_conda_forge_specific(
+    meta, recipe_dir, lints, hints, rattler_lint=False
+):
     gh = github.Github(os.environ["GH_TOKEN"])
 
-    # Retrieve sections from meta
-    package_section = get_section(meta, "package", lints)
-    extra_section = get_section(meta, "extra", lints)
-    sources_section = get_section(meta, "source", lints)
-    requirements_section = get_section(meta, "requirements", lints)
-    outputs_section = get_section(meta, "outputs", lints)
+    if not rattler_lint:
+        # Retrieve sections from meta.yaml
+        package_section = get_section(meta, "package", lints)
+        extra_section = get_section(meta, "extra", lints)
+        sources_section = get_section(meta, "source", lints)
+        requirements_section = get_section(meta, "requirements", lints)
+        outputs_section = get_section(meta, "outputs", lints)
+    else:
+        # Retrieve sections from recipe.yaml
+        package_section = meta.get("package", {})
+        extra_section = meta.get("extra", {})
+        sources_section = meta.get("source", [])
+        requirements_section = rattler_loader.load_all_requirements(meta)
+        outputs_section = meta.get("outputs", {})
```

Comment:
Can we move this logic to `get_section`?

Reply:
to be honest I think it's better to keep them explicitly in this way
If we move it to `get_section`, we will add also there `rattler_lint` and just use meta.get in case of rattler-build.
keeping them in `run_conda_forge` function seems a more transparent option, without touching other functions

wdyt?

Reply:
I'd rather that we have these in helper functions so that the main logic stays clean. Thoughts?

Reply:
Agree. 
I've moved them into get_rattler_section and get_meta_section which are under get_section umbrella.

wdyt?